ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,365 @@ def decorated(*args, **kwargs):
+    """
+    Given the job_id and action in the endpoint, perform the action on the job.
+
+    Supported actions: do_auto_resubmit, name, parallel_submit, copy, kill, force_status, resubmit, runPostProcessors, submit
+
+    Important Info:
+    1) do_auto_resubmit: User needs to provide 'do_auto_resubmit' field and it's value in the request body.
+    2) name: User needs to provide 'name' field and it's value in the request body.
+    3) parallel_submit: User needs to provide 'parallel_submit' field and it's value in the request body.
+    4) force_status: User needs to provide 'force_status' field and it's value in the request body.
+
+    :param job_id: int
+    :param action: str
+    :param attribute: str
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+
+    # Job existence check
+    if job_id not in list(jobs.ids()):
+        return jsonify({"success": False, "message": "Job with ID {} does not exist".format(job_id)}), 400
+
+    supported_actions = ["do_auto_resubmit", "name", "parallel_submit", "copy", "kill", "force_status", "resubmit",
+                         "runPostProcessors", "submit"]
+
+    # Action support check
+    if action not in supported_actions:
+        return jsonify({"success": False,
+                        "message": "Job action {} is not currently supported or does not exist".format(action)}), 400
+
+    # Function for action: do_auto_resubmit
+    if action == "do_auto_resubmit":
+
+        # Get value detail from the request data
+        value = request.form.get("do_auto_resubmit")
+
+        # Value check
+        if value is None:
+            return jsonify({"success": False,
+                            "message": "Please provide do_auto_submit value in the request body."}), 400
+
+        # Value validator
+        if value in ["True", "False", "true", "false"]:
+            value = False if value == "False" or value == "false" else True
+        else:
+            return jsonify({"success": False,
+                            "message": "Please provide do_auto_submit value as True or False."}), 400
+
+        # Change the attribute value of the Job
+        try:
+            j = jobs(job_id)
+            setattr(j, action, bool(value))
+        except Exception as err:
+            return jsonify({"success": False, "message": str(err)}), 400
+
+    # Function for action: name
+    if action == "name":
+
+        # Get value detail from the request data
+        value = request.form.get("name")
+
+        # Value check
+        if value is None:
+            return jsonify({"success": False,
+                            "message": "Please provide name value in the request body."}), 400
+
+        # Change the attribute value of the Job
+        try:
+            j = jobs(job_id)
+            setattr(j, action, str(value))
+        except Exception as err:
+            return jsonify({"success": False, "message": str(err)}), 400
```

Comment:
Can we get all the attributes from the schema of the `GangaObject`, in this case the `Job`. See https://github.com/ganga-devs/ganga/blob/92c6ac046b8b323d5428434289ee430c7dd2375a/ganga/GangaCore/GPIDev/Lib/Job/Job.py#L201 . If there is a `SimpleType` in the schema, it seems to me that we could just auto generate the code for it here.